pkg/storage/tsdb/ref_cache_test.go

```diff
@@ -96,30 +97,48 @@ func TestRefCache_Purge(t *testing.T) {
 	}
 }
 
-var series1M = prepareSeries(1e6)
+var goroutines = []int{50, 100, 500}
 
-func BenchmarkRefCacheConcurrency_1m_50(b *testing.B) {
-	benchmarkRefCacheConcurrency(b, series1M, 50)
-}
+func BenchmarkRefCacheConcurrency_single_label(b *testing.B) {
+	const seriesCount = 1e5
 
-func BenchmarkRefCacheConcurrency_1m_250(b *testing.B) {
-	benchmarkRefCacheConcurrency(b, series1M, 100)
-}
+	series := make([]labels.Labels, seriesCount)
+
+	for s := 0; s < len(series); s++ {
+		series[s] = labels.Labels{
+			{Name: "a", Value: strconv.Itoa(s)},
+		}
+	}
 
-func BenchmarkRefCacheConcurrency_1m_1000(b *testing.B) {
-	benchmarkRefCacheConcurrency(b, series1M, 500)
+	for _, num := range goroutines {
+		b.Run(fmt.Sprintf("%d", num), func(b *testing.B) {
+			benchmarkRefCacheConcurrency(b, series, num)
+		})
+	}
 }
 
-func prepareSeries(numSeries int) []labels.Labels {
-	series := make([]labels.Labels, numSeries)
+func BenchmarkRefCacheConcurrency_long_labels(b *testing.B) {
+	const seriesCount = 1e5
+	const labelsCount = 10
+	const namePrefix = "abcdefghij"
+	const valuePrefix = "pqrstuvzyx"
```

Comment:
Not sure these can be considered long. This week I've seen a production cluster several tens of thousands of series with label values longer then 1KB. Would it make sense to modify the benchmark to see how it performs with such long labels?

Reply:
I've changed it to use longer label names and values.